Jenkinsfile

```diff
@@ -63,22 +64,86 @@ pipeline {
         }
       }
     }
-    stage ('Run application tests') {
+    stage ('Build and Test') {
       when {
         expression { env.BRANCH_NAME != 'latest' }
       }
-      agent {
-        docker {
-          image "${nodeImage}"
-          label nodeName
-          args '-u root -v /etc/pki:/certs'
+      parallel {
+        stage('Test Development') {
+          agent {
+            docker {
+              image "${nodeImage}"
+              label nodeName
+              args '-u root -v /etc/pki:/certs'
+            }
+          }
+          steps {
+            sh 'make install'
+            sh 'make developmentTests'
+          }
         }
+
+        stage('Test Production') {
+          agent {
+            docker {
+              image "${nodeImage}"
+              label nodeName
+              args '-u root -v /etc/pki:/certs'
+            }
+          }
+          steps {
+            // Testing
+            sh 'make installProd'
+            sh 'make productionTests'
+          }
+        }    
       }
-      steps {
-        sh 'make install'
-        sh 'make developmentTests'
-        sh 'make installProd'
-        sh 'make productionTests'
+      post {
+        always {
+          script {
+            stageName = env.STAGE_NAME
+          }
+        }
+      }
+    }
+    stage ('Build, Test & Package') {
+      when {
+        expression { env.BRANCH_NAME == 'latest' }
+      }
+      parallel {
+        stage('Test Development') {
+          agent {
+            docker {
+              image "${nodeImage}"
+              label nodeName
+              args '-u root -v /etc/pki:/certs'
+            }
+          }
+          steps {
+            sh 'make install'
+            sh 'make developmentTests'
+          }
+        }
+
+        stage('Test Production and Zip Production') {
+          agent {
+            docker {
+              image "${nodeImage}"
+              label nodeName
+              args '-u root -v /etc/pki:/certs'
+            }
+          }
+          steps {
```

Comment:
For demonstrating that this PR works as expected, I'd like to see a job running with this packing step included. (This can be done by just changing the check on this stage only to run for non-`latest` branches) 
Then having that job linked in the PR as a comment.

Reply:
I already done this. https://ci.news.tools.bbc.co.uk/blue/organizations/jenkins/simorgh/detail/race-condition/11/pipeline